src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenSwitchedToSoleCo.java

```diff
@@ -69,17 +102,197 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<C
+        } else if (ccdAccessService.isApplicant2(httpServletRequest.getHeader(AUTHORIZATION), caseId)) {
             notificationDispatcher.send(applicant2SwitchToSoleCoNotification, data, caseId);
         }
 
+        if (ConditionalOrder.D84WhoApplying.APPLICANT_2.equals(data.getConditionalOrder().getD84WhoApplying())) {
+            switchUserRoles(caseId);
+            switchApplicantData(data);
+        }
+
         return AboutToStartOrSubmitResponse.<CaseData, State>builder()
             .data(data)
             .build();
     }
+
+    private void switchUserRoles(final Long caseId) {
+        final String auth = idamService.retrieveSystemUpdateUserDetails().getAuthToken();
+        final String s2sToken = authTokenGenerator.generate();
+        final CaseAssignmentUserRolesResource response =
+            caseAssignmentApi.getUserRoles(auth, s2sToken, List.of(caseId.toString()));
+
+        final List<CaseAssignmentUserRole> creatorUser = response.getCaseAssignmentUserRoles().stream()
+            .filter(caseAssignmentUserRole -> CREATOR.getRole().equals(caseAssignmentUserRole.getCaseRole()))
+            .limit(1)
+            .collect(toList());
+
+        final List<CaseAssignmentUserRole> applicant2User = response.getCaseAssignmentUserRoles().stream()
+            .filter(caseAssignmentUserRole -> APPLICANT_2.getRole().equals(caseAssignmentUserRole.getCaseRole()))
+            .limit(1)
+            .collect(toList());
+
+        final String currentCreatorUserId = !creatorUser.isEmpty() ? creatorUser.get(0).getUserId() : "";
+        final String currentApplicant2UserId = !applicant2User.isEmpty() ? applicant2User.get(0).getUserId() : "";
+
+        removeCaseUserRoles(caseId, auth, s2sToken, currentCreatorUserId, CREATOR);
+        removeCaseUserRoles(caseId, auth, s2sToken, currentApplicant2UserId, APPLICANT_2);
+
+        addCaseUserRoles(caseId, auth, s2sToken, currentApplicant2UserId, CREATOR);
+        addCaseUserRoles(caseId, auth, s2sToken, currentCreatorUserId, APPLICANT_2);
+    }
+
+    private void removeCaseUserRoles(final Long caseId,
+                                     final String auth,
+                                     final String s2sToken,
+                                     final String userId,
+                                     final UserRole role) {
+
+        caseAssignmentApi.removeCaseUserRoles(
+            auth,
+            s2sToken,
+            ccdAccessService.getCaseAssignmentRequest(caseId, userId, null, role)
+        );
+    }
+
+    private void addCaseUserRoles(final Long caseId,
+                                  final String auth,
+                                  final String s2sToken,
+                                  final String userId,
+                                  final UserRole role) {
+
+        caseAssignmentApi.addCaseUserRoles(
+            auth,
+            s2sToken,
+            ccdAccessService.getCaseAssignmentRequest(caseId, userId, null, role)
```

Comment:
Would it be better if there was another method without the organisationId argument so that we don't have to pass null